src/Language/PureScript/CoreImp/Optimizer.hs

```diff
@@ -37,7 +37,9 @@ optimize js = do
       [ inlineCommonValues
       , inlineCommonOperators
       ]) js
-    untilFixedPoint (return . tidyUp) . tco . inlineST =<< untilFixedPoint (return . magicDo) js'
+    untilFixedPoint (return . tidyUp) . tco . inlineST
+      =<< untilFixedPoint (return . magicDo')
+      =<< untilFixedPoint (return . magicDo) js'
```

Comment:
I know you already merged this, but since this is just a pure fn, could this not just be `untilFixedPoint (return . magicDo' . magicDo)` just to lose the extra fixed point pass (which can be expensive).

Reply:
I wasn't entirely sure if it was safe to interleave the two transformations. In the rare case someone mixes the two in the same module, but I guess the typechecker would've rejected the program at that point...